src/NzbDrone.Core/DecisionEngine/DownloadDecisionComparer.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NzbDrone.Core.Indexers;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.Profiles.Delay;
+
+namespace NzbDrone.Core.DecisionEngine
+{
+    public class DownloadDecisionComparer : IComparer<DownloadDecision>
+    {
+        private readonly IDelayProfileService _delayProfileService;
+        public delegate int CompareDelegate(DownloadDecision x, DownloadDecision y);
+        public delegate int CompareDelegate<TSubject, TValue>(DownloadDecision x, DownloadDecision y);
+
+        public DownloadDecisionComparer(IDelayProfileService delayProfileService)
+        {
+            _delayProfileService = delayProfileService;
+        }
+
+        public int Compare(DownloadDecision x, DownloadDecision y)
+        {
+            var comparers = new List<CompareDelegate>
+            {
+                CompareQuality,
+                CompareProtocol,
+                CompareEpisodeCount,
+                CompareEpisodeNumber,
+                ComparePeers,
+                CompareAge,
+                CompareSize
+            };
+
+            return comparers.Select(comparer => comparer(x, y)).FirstOrDefault(result => result != 0);
+        }
+
+        private int Compare<TSubject, TValue>(TSubject left, TSubject right, Func<TSubject, TValue> funcValue)
```

Comment:
CompareBy feels a bit awkward to me, but I don't have a strong objection to it and https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.comparisons/compare-by.html, so done.